### Домашнее задание №5 ###

**Проверка статистических гипотез. P-значения. Доверительные интервалы. A/B-тестирование**

**1. Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16. Найти доверительный интервал для оценки математического ожидания с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256**

Имеем по условию задачи:<br>
$\sigma = 16$<br>
$M = 80$<br>
$n = 256$

$p = 1-\alpha = 0,95$ соответсвенно получается $\alpha = 0,05$

Нижняя и верхняя границы доверительного интервала: $T_{1,2} = \overline{X} \pm Z_{\frac{\alpha}{2}} \cdot \frac{\sigma}{\sqrt{n}}$

Если известно среднее квадратическое отклонение генеральной совокупности, то используем Z-критерий.

Согласно Z-таблице:

### $Z_{\frac{a}{2}} = Z_{0,025} = -1.96$

Расчитаем оп формуле доверительный интервал:<br>
$T_{1,2} = 80 \pm 1,96 \cdot \frac{16}{\sqrt{256}} = 80 \pm 1,96 $<br>
$T_{1} = 80 - 1,96 = 78,04$ <br>
$T_{2} = 80 + 1,96 = 81,96$ <br>


***Ответ: $T_{1} = 78,04$; $T_{2} = 81,96$ <br>***

**2. В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1. Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.**

Имеем по условию задачи:<br>
$n = 10$<br>
$p = 1-\alpha = 0,95$ соответсвенно получается $\alpha = 0,05$<br>
$\sigma$ у нас не известно то используем формулу $T_{1,2} = \overline{X}\pm t_{\frac{\alpha}{2}} \cdot \frac{\sigma_{н}}{\sqrt{n}}$  где $\sigma_{н}$ - несмещенная

In [1]:
import numpy as np
X = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
X

array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])

Выборочное среднее арифметическое $\overline{X}$:

In [2]:
x = X.mean()
x

6.590000000000001

Расчитаем не смещенное среднее квадратичное отклонение по выборке 𝑠0 :

In [3]:
s0=np.std(X, ddof=1)
s0

0.4508017549014448

Согласно t-таблице, для  𝛼=0,025  и числа степеней свободы равному  𝑛−1=9  значение  𝑡=2,262 <br>

In [4]:
t1 = x - 2.262*s0/np.sqrt(10)
t1

6.267538255912426

In [5]:
t2 = x + 2.262*s0/np.sqrt(10)
t2

6.912461744087575

***Ответ: $T_{1}=6.2675$; $T_{2}=6.9125$***

**3. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5мм, а дисперсия известна и равна 4 кв.мм.**

Имеем по условию задачи:<br>
$\mu_0 = 17$<br>
$\mu = 17.5$<br>
n=100<br>
α=0.05<br>
$\sigma^2=4 кв.мм.$<br>



Выдвинем гипотезы:<br>
$H_0: \, \mu = \mu_0$<br>
$H_1: \, \mu > \mu_0$

Критерий  Z –тест:<br>



## $Z_н = \frac{\overline{X} - \mu_0}{\frac{\sigma}{\sqrt{n}}}=\frac{17,5 - 17}{\frac{\sqrt{4}}{\sqrt{100}}} = 2,5$ ##

Согласно Z-таблице:<br>
$Z_T = Z_{0,95} = 1,65$

Так как $Z_н > Z_T$ то гипотеза $H_0$ не верна

***Ответ: утверждение, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм неверно при $\alpha=0,05$***

**4. Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?**

Имеем по условию задачи:<br>
$\mu_0 = 200$<br>
n=10<br>
α=1%<br>


In [6]:
X = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
x=X.mean()

Среднее квадратичное отклонение по выборке (несмещенное) $s_0$:

In [7]:
s0=np.std(X, ddof=1)

Выдвинем гипотезы:<br>
$H_0: \, \mu = \mu_0$<br>
$H_1: \, \mu > \mu_0$

Вычислим t критерий:


# $t_р = \frac{\mu - \mu_0}{\frac{s_0}{\sqrt{n}}}$ #

In [8]:
tp=(x - 200)/(s0/np.sqrt(10))
tp

-1.0651074037450896

Согласно t-таблице, для $\frac{\alpha}{2} = 0,005$ и числа степеней свободы равному $n-1 = 9$ значение $t_T = 3,25$

Так как $t_р$ попадает в интервал от $-t_T$ до $t_T$, то гипотеза $H_0$ верна

***Ответ: утверждение проддовца, что средний вес пачки печенья составляет 200 г., верно***

____

Приложение: Z-таблицы

![title](image1.png) ![title](image2.png)  ![title](image3.png)